src/include/cluster/dialogs.rb

```diff
@@ -764,9 +797,35 @@ def ValidateCorosyncQdevice
       end
 
       if UI.QueryWidget(Id(:corosync_qdevice), :Value) && Cluster.memberaddr.size <= 0
+        # Intent not return false since address is in another dialog.
         Popup.Message(_("Member Address is required when enable corosync qdevice"))
       end
 
+      if !["off"].include?(UI.QueryWidget(Id(:heuristics_mode), :Value))
```

Comment:
This is a little bit weird, why not simply

```ruby

if UI.QueryWidget(Id(:heuristics_mode), :Value) != "off"

```

Although as you use it again later, I'd use a bunch of private methods for those _heuristics queries_, I mean

```ruby
...
private
...
def heuristics_mode
 UI.QueryWidget(Id(:heuristics_mode), :Value)
end

def heuristics_sync_timeout
  UI.QueryWidget(Id(:heuristics_sync_timeout), :Value).to_i
end

...

```

which could help to simplify those `if` statements

Reply:
Good point. It used to `!["off", "sync"]` in the beginning, i simply removed the "sync"...

Also good idea to using private function for the long query. Considering it only query once in that `function` and `if` statements. Quite many querys in this module like this, can i just leave as it is for a unified (not decent) format?

Reply:
Partial fixed in the new commit.